### ISSUE TITLE
List optional dependencies in `peerDependencies` + `peerDependenciesMeta`

### ISSUE THREAD

**Issue by moshest** [WebReflection/linkedom#249]
For example in `package.json`:
```json
{
  "name": "linkedom",
  "version": "0.16.1",
  "peerDependencies": {
    "canvas": ">= 2",
  },
  "peerDependenciesMeta": {
    "canvas": {
      "optional": true
    }
  }
}
```

Please see docs here: https://docs.npmjs.com/cli/v7/configuring-npm/package-json#peerdependenciesmeta

**Comment by WebReflection** [comment]
this "*smells like a PR*" 😄

**Comment by jlara005** [comment]
When is a fix for the warnings going to be released? I'm on latest 0.18.5

WARNING in ./node_modules/linkedom/commonjs/perf_hooks.cjs 3:24-45
Module not found: Error: Can't resolve 'perf_hooks' in 'C:\Dev\TestProject\node_modules\linkedom\commonjs'
 @ ./node_modules/linkedom/cjs/interface/document.js 2:22-62
 @ ./node_modules/linkedom/cjs/index.js 3:30-64

**Comment by WebReflection** [comment]
> When is a fix for the warnings going to be released?

When the PR lands

**Comment by WebReflection** [comment]
@rxliuli thanks for the hint ... it looks like that file could just go away entirely at this point as NodeJS 16 is long-time-ago LTS ... and if that's the case, I wonder if the other PR would still be needed or not 🤔

**Comment by rxliuli** [comment]
> [@rxliuli](https://github.com/rxliuli) thanks for the hint ... it looks like that file could just go away entirely at this point as NodeJS 16 is long-time-ago LTS ... and if that's the case, I wonder if the other PR would still be needed or not 🤔

I haven't encountered any peerDependencies related issues, just a warning when building with vite, which was resolved by no longer importing perf_hooks.
https://vite.dev/guide/troubleshooting.html#module-externalized-for-browser-compatibility

 I was responding to @jlara005's question, and I'm not sure what that PR is doing.

**Comment by WebReflection** [comment]
@rxliuli fair enough ... PR to remove `performance` welcomed but I might do it myself soon as it's always a pleasure to drop legacy try catches and trust modern environments instead ... although your quick and easy shim around it might still be needed to avoid breaking very old projects that carelessly use latest linkedom whatever happens.

**Comment by rxliuli** [comment]
@WebReflection I created a small PR to fix this issue. If you're willing to merge and release a minor version, I can remove my local patch.
https://github.com/WebReflection/linkedom/pull/311

**Comment by rxliuli** [comment]
@WebReflection  I just want to let you know that the new version requires explicitly pnpm installing the canvas package, otherwise it will throw an error, while previous versions didn't have this error.

```js
Uncaught Error: Could not resolve "canvas" imported by "linkedom". Is it installed?
```

**Comment by WebReflection** [comment]
@rxliuli does `npm` or *bun* complain? I can't write apparently a package for all runtimes ... but if peerDependency is the way to go, I'd rather ask you to file an issue to pnpm instead ...
